src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -5064,13 +5064,39 @@ int RGWRados::check_bucket_empty(const DoutPrefixProvider *dpp, RGWBucketInfo& b
 
   return 0;
 }
+
+int RGWRados::store_delete_bucket_info_flag(RGWBucketInfo& bucket_info, std::map<std::string, bufferlist>& attrs, optional_yield y, const DoutPrefixProvider *dpp) {
+  const rgw_bucket& bucket = bucket_info.bucket;
+  static constexpr auto max_retries = 10;
+  int retries = 0;
+  int r = 0;
+  do {
+    bucket_info.flags |= BUCKET_DELETED;
+    r = ctl.bucket->store_bucket_instance_info(bucket, bucket_info, y, dpp, RGWBucketCtl::BucketInstance::PutParams()
```

Comment:
should this add a 'deleted' log layout too?

Reply:
may be we could check if all the generations have been fully trimmed on the peer zones before calling metadata removal on master zone in the trim logic?

Reply:
> may be we could check if all the generations have been fully trimmed on the peer zones before calling metadata removal on master zone in the trim logic?

huh no, if the bucket metadata doesn't exist, there will be nothing left to check. wonder what the StatusShards would return in that case.

Reply:
> ok but if trim runs on the master zone first and clears the bucket metadata, that would end up syncing this metadata delete op even before we trim the bilog on the other zones.

this will only happen if we add the delete op to datalog entry.

Reply:
> huh no, if the bucket metadata doesn't exist, there will be nothing left to check. StatusShards should return -ENOENT in that case.

we may need some special handling in this case, in bilog trim logic.